doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+coding conventions, creating your environment, picking an issue to fix, and
+opening a pull request, please read our full `Development Workflow
+<https://github.com/sympy/sympy/wiki/Development-workflow>`_ guide.
+
+Best Practices for Writing Documentation
+========================================
+
+Please follow these formatting, style, and tone preferences when writing
+documentation.
+
+Formatting Preferences
+----------------------
+
+**Math vs. Code**
+
+**Math:** text that is surrounded by dollar signs $ _ $ will be rendered as
+LaTeX math. Any text that is meant to appear as LaTeX math should be written as
+$math$. In the HTML version of the docs, MathJax will render the math.
+
+**Example**
+
+::
+
+  The Bessel $J$ function of order $\nu$ is defined to be the function
+  satisfying Bessel’s differential equation.
+
+**Code:** text that should be printed verbatim, such as code, should be
+surrounded by a set of double backticks ``like this``. (WORK IN PROGRESS: change
+single backticks to render as code via issue `#13519
+<https://github.com/sympy/sympy/issues/13519>`_.)
+
+**Example**
+
+::
+
+  To use this class, define the ``_rewrite`` and ``_expand`` methods.
+
+Sometimes a variable will be the same in both math and code, and can even
+appear in the same paragraph, making it difficult to know if it should be
+formatted as math or code. If the sentence in question is discussing
+mathematics, then LaTeX should be used, but if the sentence is discussing the
+SymPy implementation specifically, then code should be used.
+
+**Example**
+
+::
+
+  def stirling(n, k, d=None, kind=2, signed=False):
+      """
+      ...
+
+      The first kind of Stirling number counts the number of permutations of
+      ``n`` distinct items that have ``k`` cycles... This counts the ways to
+      partition $n$ consecutive integers into $k$ groups with no pairwise
+      difference less than $d$.
+
+      """
+
+Note that in the above example, the first instances of *n* and *k* are
+referring to the input parameters of the function ``stirling``, so they refer
+to Python variables and are therefore formatted as code. The second instances
+of *n* and *k* are talking about mathematical expressions, so they are
+formatted as math.
+
+In general, the rule of thumb is to consider if the variable in question were
+something that rendered differently in code and in math. For example, the Greek
+letter α would be written as ``alpha`` in code and $\alpha$ in LaTeX. The
+reason being that $\alpha$ cannot be used in contexts referring to Python code
+because it is not valid Python, and conversely ``alpha`` would be incorrect in
+a math context because it does not render as the Greek letter (α).
+
+Style Preferences
+-----------------
+
+**Spelling and Punctuation**
+
+All narrative writing in SymPy follows American spelling and punctuation
+standards. For example, “color” is preferred over “colour” and commas should be
+placed inside of quotation marks.
```

Comment:
Is commas-inside-quotes an American punctuation standard? Being a British-English native speaker I've never noticed that before...

Reply:
Yes, in American editorial/printer standards punctuation always occurs inside of closing quotation marks and parentheses, with the exception of certain instances when the punctuation belongs not to the quoted or parenthetical material but to the larger sentence.